### PR TITLE
Allow the user agent header to be set via a preference

### LINKED ISSUE #123: Allow setting User-Agent header via preferences
Currently, the default value of the User-Agent header is set to `HTTP-REPL`  here:
https://github.com/aspnet/HttpRepl/blob/b208ca44a4f9361027b42df89a79594c291b72a5/src/Microsoft.HttpRepl/HttpState.cs#L45

You can override it using the various methods of setting headers, but it would be nice to have this pulled from preferences so if you, for instance, always wanted to be emulating a particular User-Agent, you could have that persist.

### PR DESCRIPTION
**Pull request by tlmii**
This resolves #123. 

Currently, since there's no mechanism to notify other parts of the system when a preference changes, this will require a restart. I put in #165 to address that in the future for this and other settings that it may be useful for.

### CODE REVIEW

File: src/Microsoft.HttpRepl.Tests/HttpStateTests.cs [hunk 1]
```diff
@@ -188,14 +188,46 @@ public void GetEffectivePathForPrompt_NullBaseAddress_ReturnsNull()
...
+                fileSystem = new MockedFileSystem();
+            }
+            else
+            {
+                fileSystem = new FileSystemStub();
+            }
+            UserFolderPreferences preferences = new UserFolderPreferences(fileSystem, userProfileDirectoryProvider, null);
+            if (preferencesFileContent != null)
+            {
+                ((MockedFileSystem)fileSystem).AddFile(preferences.PreferencesFilePath, preferencesFileContent);
```
Comment (jimmylewis) [inline src/Microsoft.HttpRepl.Tests/HttpStateTests.cs]: Could this happen in the block above?  Seems to keep the code a little simpler instead of having two checks on `preferencesFileContent`.

Comment (tlmii) [reply src/Microsoft.HttpRepl.Tests/HttpStateTests.cs]: It needs the preferences. Which in turn needs the filesystem. I agree its ugly.

File: src/Microsoft.HttpRepl/Preferences/WellKnownPreference.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 using System.Collections.Generic;
 using System.Reflection;
+using System.Security;
```
Comment (jimmylewis) [inline src/Microsoft.HttpRepl/Preferences/WellKnownPreference.cs]: why is this needed?

Comment (tlmii) [reply src/Microsoft.HttpRepl/Preferences/WellKnownPreference.cs]: Its not. Nor do I know why it was there. But its gone now.
